chapters/mime-type/index.rst

```diff
@@ -0,0 +1,119 @@
+
+Manual mime type assignment
+---------------------------
+To have the full control over the mime type, it is also possible to set it manually like shown in the following example:
+
+.. confval:: manual_assignment.py
+
+    .. highlight:: python
+    .. code-block:: python
+
+        from jina import Document
+
+        svg_content = (
+            '<svg height="100" width="100">'
+            '  <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />'
+            '</svg>'
+        ).encode('utf8')
+        d = Document()
+        d.buffer = (svg_content)
+        assert d.mime_type == 'image/svg'
+        # the mime type can be overwritten
+        d.mime_type = 'text/plain'
+        # assigning an invalid mime type leads to a ``ValueError``
+        d.mime_type = 'invalid/type' # raises exception
+
+Mime type in chunks
+-------------------
+Chunks are created by ``Segmenters``.
+For a variety of use cases, chunks have the same mime type as their parent documents.
+Examples are text documents where each sentence (chunk) is a text document on its own.
+Also when segmenting images or audio, most applications are creating chunks of the same mime type.
+However, there are scenarios where it makes sense to change the mime type on chunk level.
+A common examples would be for instance video documents where the chunks are images.
+The ``Segmenters`` are responsible for assigning the right mime type to the chunks they create.
+In case no mime type is set, the ``SegmentDriver`` assigns the mime type of the parent document as default value.
+The following example illustrates a simple segmenter, which sets the ``mime_type`` for each chunk it creates.
+
+.. confval:: dummy_segmenter.py
+
+    .. highlight:: python
+    .. code-block:: python
+
+        from jina.executors.segmenters import BaseSegmenter
+
+        class DummySegmenter(BaseSegmenter):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+            def segment(self, text: str, *args, **kwargs):
+                results = [{
+                    'text': word,
+                    'mime_type': 'text/plain'
+                } for word in text.split()]
+                return results
+
+
+Usage in driver
+---------------
+Drivers can access the mime type of the documents in order to handle them accordingly.
+The following driver only encodes documents where the ``mime_type`` is ``'text/plain'``:
+
+.. confval:: special_segment_driver.py
+
+    .. highlight:: python
+    .. code-block:: python
+
+        from ..types.document import Document
+        from .. import DocumentSet
+
+         class EncodeDriver(FastRecursiveMixin, BaseEncodeDriver):
+            """Extract the content from documents and call executor and do encoding
+            """
+
+            def _apply_all(self, leaves: Iterable['DocumentSet'], *args, **kwargs) -> None:
+                docs = DocumentSet.flatten(leaves)
```

Comment:
this code snippet is going to be outdated by redesign of Driver. I think if the point is to explain `mime_type`, better to simplify the code to show things about the `mime_type` without the rest of the noise.

Reply:
Valid point!
Do you have an idea what feature of the mime_type I should promote?
At the moment I could not find any other usage. The driver case was somehow made up by me.

Reply:
No, I think u explained everything there is to say very well.